tensorflow/contrib/tensorrt/segment/segment.cc

```diff
@@ -430,6 +436,11 @@ tensorflow::Status SegmentGraph(
   std::vector<UnionFind<SimpleNode*>> node_segments;
   for (int i = 0; i < graph->num_node_ids(); ++i) {
     SimpleNode* node = graph->FindNodeId(i);
+    const string requested_device =
```

Comment:
@aaroey do you know the difference between requested devices and assigned devices? Which one has a higher priority?

We need to specify our expectation precisely. 
- If user requests CPU for an op, should we include it in a segment? 
- If user doesn't specify CPU but TF assigns CPU to an op (our new NMS op is like this case), should we include the op in a segment? 
- If user requests GPU, but TF assigns CPU, should we include the op in a segment?

Reply:
From my observations, nodes all have a blank assigned_device during the segmentation stage. The requested device is set whether using `with tf.device()` or not. So there is no way to distinguish between a user specified device or what TF requests by default.

I am removing the segmenter rule to exclude CPU ops because in some cases an op may be assigned to CPU but is supported by TRT and we want to include the op in that case.